### PR TITLE
JAVA-5683: Difference Between List.of and Arrays.asList

### PR DESCRIPTION
**Pull request by harry9656**
Added compatibility to the latest LTS Java 17

### CODE REVIEW

**Comment by lor6** [comment]
retest this please

File: core-java-modules/core-java-collections-list-4/pom.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0"
-    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-collections-list-4/pom.xml:4]: Let's undo this change

File: core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ArraysAsListUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class ArraysAsListUnitTest {
+
+@Test
+void shouldThrowUnsupportedOperationExceptionWhenAddingElementsToFixedList() {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ArraysAsListUnitTest.java:13]: These need to be indented

Comment (ashleyfrieze) [inline core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ArraysAsListUnitTest.java:13]: Test naming should be in `givenWhenThen` format

File: core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ArraysAsListUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    List<Integer> list = Arrays.asList(1, 2, 3, 4, 5);
+    assertThrows(UnsupportedOperationException.class, () -> list.add(6));
+}
+
+@Test
+void shouldThrowUnsupportedOperationExceptionWhenRemovingElementsFromFixedList() {
+    List<Integer> list = Arrays.asList(1, 2, 3, 4, 5);
+    assertThrows(UnsupportedOperationException.class, () -> list.remove(1));
+}
+
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ArraysAsListUnitTest.java:23]: I'd expect to see unit tests for the contents of the list in the positive case, as well as its exception throwing

File: core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ListOfUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+@Test
+void shouldThrowUnsupportedOperationExceptionWhenChangingElementsFromImmutableList() {
+    List<String> list = List.of("one", "two", "three");
+    assertThrows(UnsupportedOperationException.class, () -> list.set(1, "four"));
+}
+
+@Test
+void shouldThrowNullPointerExceptionWhenCreatingListWithNullElement() {
+    assertThrows(NullPointerException.class, () -> List.of("one", null, "two"));
+}
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ListOfUnitTest.java:32]: Indentation

Comment (ashleyfrieze) [reply core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ListOfUnitTest.java:32]: And I'd expect to see tests for the case of having the list.

**Review by ashleyfrieze — CHANGES_REQUESTED** [review]

**Comment by harry9656** [comment]
@ashleyfrieze I made the changes you requested and added a test to verify the content of the List.

File: core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ArraysAsListUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        List<Integer> list = Arrays.asList(array);
+        assertThat(list).containsExactly(1, 2, 3, 4);
+    }
+
+    @Test
+    void givenAnListOfStringCreatedUsingArraysAsListWhenChangingTheOriginalArrayTheReturnListWillAlsoChange() {
+        Integer[] array = new Integer[]{1, 2, 3};
+        List<Integer> list = Arrays.asList(array);
+        array[0] = 1000;
+        assertEquals(1000, list.get(0));
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ArraysAsListUnitTest.java:26]: Personally, I'd prefer to see `assertThat` used throughout these tests, but this is valid

File: core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ArraysAsListUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    @Test
+    void givenAnListOfStringCreatedUsingArraysAsListWhenChangingTheReturnedListTheOriginalArrayWillAlsoChange() {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ArraysAsListUnitTest.java:30]: This is a list of `Integer`.

**Review by ashleyfrieze — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ArraysAsListUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    void givenAnArrayOfIntegersCreatedUsingArraysAsListWhenModifyingAnExistingElementThenModifyIt() {
+        List<Integer> list = Arrays.asList(1, 2, 3, 4);
+        list.set(1, 1000);
+        assertThat(list.get(1)).isEqualTo(1000);
+    }
+
+    @Test
+    void givenAnArrayCreatedWithArraysAsListWhenAddingANewElementThenUnsupportedOperationExceptionIsThrown() {
+        List<Integer> list = Arrays.asList(1, 2, 3, 4, 5);
+        assertThrows(UnsupportedOperationException.class, () -> list.add(6));
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-collections-list-4/src/test/java/com/baeldung/list/creation/ArraysAsListUnitTest.java:46]: Don't change it, because it's fine, but note you can also use `assertThatThrownBy(...)` here - e.g. `assertThatThrownBy(() -> list.add(6)).isInstanceOf(UnsupportedOperationException.class)`

**Review by ashleyfrieze — APPROVED** [review]
